src/components/users/EditProfileForm.js

```diff
@@ -1,12 +1,171 @@
+      alert(
+        intl.formatMessage({
+          id: 'editprofile.success.text',
+        })
+      );
+      setEditProfile(null);
+      dispatch({ type: userConstants.USER_UPDATE_SUCCESS_FINISHED });
+    }
+  }, [updated, setEditProfile, intl, dispatch]);
+
+  const handleChange = ({ target }) => {
+    setIsSubmitted(false);
+    const { name, value } = target;
+    setInputs((inputs) => ({ ...inputs, [name]: value }));
+  };
+
+  const handleChangeGender = (selectGender) => {
+    setIsSubmitted(false);
+    setInputs((inputs) => ({
+      ...inputs,
+      gender: selectGender['value'],
+      selectGender: selectGender,
+    }));
+  };
+
+  const changedNameOrGender =
+    (name && name !== user.name) || gender !== user.gender;
+
+  const changedPassword = password && equalPasswords;
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setIsSubmitted(true);
+    if (changedNameOrGender || changedPassword) {
+      dispatch(userRequest());
+      dispatch(
+        userActions.update(
+          name,
+          gender,
+          password,
+          confirmNewPassword,
+          changedNameOrGender
+        )
+      );
+    }
+  };
+
+  return (
+    <form align="center" className="edit-profile-form" onSubmit={handleSubmit}>
+      <FormInput
+        labelClassName="user-form__text"
+        inputClassName="user-form__input"
+        inputLabel={intl.formatMessage({
+          id: 'editprofile.update.name.text',
+        })}
+        inputType="name"
+        inputName="name"
+        inputValue={name}
+        inputOnChange={handleChange}
+        error={isSubmitted && !name}
+        errorMsg={intl.formatMessage({
+          id: 'editprofile.empty.name.text',
+        })}
+      />
+      <p className="user-form__text">
+        {intl.formatMessage({
+          id: 'editprofile.update.gender.text',
+        })}
+      </p>
+      <div className="edit-profile-form__gender-select">
+        <FormSelect
+          customStyle={genderSelectStyle}
+          customClassName="user-form__text user-form__select"
+          optionsSet={genders}
+          onChangeFunction={handleChangeGender}
+          valueSelect={selectGender}
+        />
+      </div>
+      <FormInput
+        labelClassName="user-form__text"
+        inputClassName="user-form__input"
+        inputLabel={intl.formatMessage({
+          id: 'editprofile.update.password.text',
+        })}
+        inputType="password"
+        inputName="password"
+        inputValue={password}
+        inputOnChange={handleChange}
+      />
+      <FormInput
+        labelClassName="user-form__text"
+        inputClassName="user-form__input"
+        inputLabel={intl.formatMessage({
+          id: 'editprofile.update.password.confirm.text',
+        })}
+        inputType="password"
+        inputName="confirmNewPassword"
+        inputValue={confirmNewPassword}
+        inputOnChange={handleChange}
+        error={!!(isSubmitted && password && !equalPasswords)}
```

Comment:
### 
```suggestion
        error={isSubmitted && password && !equalPasswords}
```

Reply:
I had to put this because in the shape I set `error` as a bool, and if I don't put `!!` the throws an error